src/Compiler/Checking/CheckPatterns.fs

```diff
@@ -104,20 +130,28 @@ and TcSimplePat optionalArgsOK checkConstraints (cenv: cenv) ty env patEnv p =
 
         match p with
         // Optional arguments on members
-        | SynSimplePat.Id(_, _, _, _, true, _) -> UnifyTypes cenv env m ty (mkOptionTy g ctyR)
+        | SynSimplePat.Id(_, _, _, _, true, _) ->
+            let optionalParamTy =
```

Comment:
Affectively the same block of code as on line 114 ?

Reply:
Pretty much. Decided to not extract it to function, so it's easier to see what the change is.